pkg/build/api/v1/types.go

```diff
@@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
 
 	// Git is the git information if the Type is BuildSourceGit
 	Git *GitInfo `json:"git,omitempty"`
+
+	// Env contains additional environment variables you want to pass into a builder container
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
> Do we call this "Env" in any other place?

yes: https://github.com/openshift/origin/blob/master/pkg/build/api/v1/types.go#L392-L393

> I would sort of expect the webhook trigger config to be able to limit what is accepted here

interesting idea.  no there is currently no way to constrain that.  I could see there being a case for a AllowEnv field on the trigger definition in the buildconfig.